### PR TITLE
Update LambdaRank example documentation

### PR DESCRIPTION
**Pull request by malcolmgreaves**
- Add reference to documentation on the query data format.
- Refer readers to the official install instructions.
- Change command to use relative path to the `lightgbm` binary built at
the project's root when following build instructions.

### CODE REVIEW

File: examples/lambdarank/README.md [hunk 1]
```diff
@@ -3,15 +3,15 @@ LambdaRank Example
...
 
 Training
 --------
 
 Run the following command in this folder:
 
-```
-"./lightgbm" config=train.conf
+```bash
+../../lightgbm config=train.conf
```
Comment (StrikerRUS) [inline examples/lambdarank/README.md]: `"` were used here to make this command cross-platform, i.e. for successful run on Windows. Please put them back.

File: examples/lambdarank/README.md [hunk 2]
```diff
@@ -21,6 +21,10 @@ You should finish training first.
 
 Run the following command in this folder:
 
+```bash
+../../lightgbm config=predict.conf
```
Comment (StrikerRUS) [inline examples/lambdarank/README.md]: Ditto for `"`.

File: examples/lambdarank/README.md [hunk 3]
```diff
@@ -21,6 +21,10 @@ You should finish training first.
...
 ```
-"./lightgbm" config=predict.conf
-```
+
+Data Format
+-----------
```
Comment (StrikerRUS) [inline examples/lambdarank/README.md]: Separate heading from text with one blank line, like it's done for `Training` section.

File: examples/lambdarank/README.md [hunk 4]
```diff
@@ -21,6 +21,10 @@ You should finish training first.
...
+To learn more about the query format used in this example, check out the [query data format](https://github.com/Microsoft/LightGBM/blob/master/docs/Parameters.rst#query-data).
```
Comment (StrikerRUS) [inline examples/lambdarank/README.md]: Use link to docs site `https://lightgbm.readthedocs.io/`, as you done for installation instructions.

Comment (malcolmgreaves) [reply examples/lambdarank/README.md]: Hmmm, but this format is not documented on the documentation website. Unless I missed something? I only found https://lightgbm.readthedocs.io/en/latest/Quick-Start.html#weight-and-query-group-data, but this section doesn't actually explain the format and it links to https://github.com/Microsoft/LightGBM/blob/master/docs/Parameters.rst#query-data.

Is there a better section on the read the docs website? Or, is there a way I could update the documentation website? (I don't know if I can from a PR?)

Comment (StrikerRUS) [reply examples/lambdarank/README.md]: https://github.com/Microsoft/LightGBM/blob/master/docs/Parameters.rst#query-data == https://lightgbm.readthedocs.io/en/latest/Parameters.html#query-data

Comment (malcolmgreaves) [reply examples/lambdarank/README.md]: Ok, got it! :D

File: examples/lambdarank/README.md [hunk 5]
```diff
@@ -21,6 +22,13 @@ You should finish training first.
...
 ```
-"./lightgbm" config=predict.conf
-```
+
+Data Format
+-----------
+
+To learn more about the query format used in this example, check out the 
+[query data format](https://lightgbm.readthedocs.io/en/latest/Parameters.html#query-data).
+
```
Comment (StrikerRUS) [inline examples/lambdarank/README.md]: Remove one excess empty line.

Comment (StrikerRUS) [reply examples/lambdarank/README.md]: OK, never mind. I'll do it in next PR with updates for the rest examples.

**Pull request merged**